### PR TITLE
 feat: create custom_form_options api

### LINKED ISSUE #6407: Implement schema and APIs to support complex custom forms
Parent issue: #6191 

Adds necessary APIs, migration files to begin implementing custom forms

### PR DESCRIPTION
**Pull request by CosmicCoder96**
Fixes #6407 

#### Changes proposed in this pull request:

- Modify Custom form schema according to #6191
- Introduce custom form options

### CODE REVIEW

**Comment by CosmicCoder96** [comment]
To be reviewed, after I test the functionality while developing client side logic on FE in parallel, to be in WIP until then.

**Comment by CosmicCoder96** [comment]
To be reviewed, after I test the functionality while developing client side logic on FE in parallel, to be in WIP until then.
Migration file missing intentionally, still working on some fields.

File: migrations/versions/rev-2019-08-24-20:32:00-ebfe89366d48_.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    op.create_table('custom_form_options',
+                    sa.Column('deleted_at', sa.DateTime(timezone=True), nullable=True),
+                    sa.Column('id', sa.Integer(), nullable=False),
+                    sa.Column('value', sa.String(), nullable=False),
+                    sa.Column('custom_form_id', sa.Integer(), nullable=True),
+                    sa.ForeignKeyConstraint(['custom_form_id'], ['custom_forms.id'], ondelete='CASCADE'),
+                    sa.PrimaryKeyConstraint('id')
+                    )
+    op.add_column('custom_forms', sa.Column('description', sa.String(), nullable=True))
+    op.add_column('custom_forms', sa.Column('is_complex', sa.Boolean(), nullable=True))
```
Comment (iamareebjamal) [inline migrations/versions/rev-2019-08-24-20:32:00-ebfe89366d48_.py]: Make this non nullable and default to false

**Comment by CosmicCoder96** [comment]
@iamareebjamal made the changes.

**Pull request merged**